atuin/src/shell/atuin.zsh

```diff
@@ -9,6 +9,15 @@
 # Source this in your ~/.zshrc
 autoload -U add-zsh-hook
 
+
+# If zsh-autosuggestions is installed, configure it to use Atuin's search. If
+# you'd like to override this, then add your config after the $(atuin init zsh)
+# in your .zshrc
+_zsh_autosuggest_strategy_atuin_top() {
+    suggestion=$(atuin search --cmd-only --limit 1 --search-mode prefix $1)
+}
+ZSH_AUTOSUGGEST_STRATEGY=atuin_top
```

Comment:
It looks like this is actually an array, we should probably append to the front: https://github.com/zsh-users/zsh-autosuggestions/blob/c3d4e576c9c86eac62884bd47c01f6faed043fc5/src/config.zsh#L18

We may also want to change the name to just `atuin` instead of `atuin_top`?

```suggestion
ZSH_AUTOSUGGEST_STRATEGY=("atuin_top" "${ZSH_AUTOSUGGEST_STRATEGY[@]}")
```